### DISCUSSION TITLE
Transforms in Sprinkles

### DISCUSSION THREAD

**Discussion by Ghirigoro** [vanilla-extract-css/vanilla-extract#819] [Q&A]
It's not clear to me how you would define transforms (or anything else that both has a large number of potential values and doesn't map cleanly to a css property). For example I can do something like this:
```ts
const translation = createVar();
const transforms = defineProperties({
  properties: {
      transform: {
        translateX1: {
          vars: { [translation]: "1px" },
          transform: `translateX(${translation})`,
        },
    },
  }
});
```
But since I have to map the var values and names it gets prohibitively big really fast (especially when you start folding in scale, rotation etc.) Am I missing a cleaner approach or is this just something that has to be worked around when using sprinkles?

**Comment by alex-mcgovern** [comment]
I think the thought behind the Sprinkles API is that it should satisfy ~80% of your styling needs, but I have still needed to write custom classes for bespoke use cases.

Something that you may find useful is a third party wrapper component around VE Sprinkles called [desert box](https://github.com/TheMightyPenguin/dessert-box). This allows you to create a box component that accepts your sprinkles args as props. Importantly it also allows override values by prefixing the sprinkles key with `__`, e.g.

```
<Box 
    __transform="translateX(50%)"
>
    content
</Box>
````
